src/Robo/RoboFileAbstract.php

```diff
@@ -101,149 +130,35 @@ public function testBehat($opts = ['silent|s' => false])
-     * @param array  $jsonContent
-     *
-     * @return array
-     */
-    protected function addErrorsToSwaggerJsonContent($applicationName, array $jsonContent)
-    {
-        $usedErrorCodes = [];
-
-        foreach ($jsonContent['paths'] ?? [] as $path => $methods) {
-            foreach ($methods as $method => $methodContent) {
-                $errorCodes = array_merge(
-                    $this->getImpliedApiErrorCodes($applicationName, $method),
-                    $methodContent['x-errors'] ?? []
-                );
-
-                foreach ($errorCodes as $errorCode) {
-                    $usedErrorCodes[$errorCode] = 1;
-
-                    // This status code is already documented, don't override
-                    if (isset($methodContent['responses'][(string)$errorCode])) {
-                        continue;
-                    }
-
-                    $methodContent['responses'][(string)$errorCode] = [
-                        'description' => $this->getApiErrorDocDescriptionByErrorCode($errorCode),
-                        'schema' => [
-                            '$ref' => '#/definitions/Error' . $errorCode,
-                        ]
-                    ];
-                }
-                if (isset($methodContent['x-errors'])) {
-                    unset($methodContent['x-errors']);
-                }
-
-                $jsonContent['paths'][$path][$method] = $methodContent;
-            }
-        }
-
-        foreach ($usedErrorCodes as $errorCode => $value) {
-            if (!isset($jsonContent['definitions']['Error' . $errorCode])) {
-                $jsonContent['definitions']['Error' . $errorCode] = $this->getApiDocErrorResponseDefinitionForCode(
-                    $errorCode
-                );
-            }
-        }
-
-        return $jsonContent;
-    }
+        switch ($this->getSwaggerVersion()) {
+            case self::SWAGGER_VERSION_V2:
+                $generator     = new SwaggerGenerator();
+                $generatorPath = 'vendor/bin/swagger';
+                break;
 
-    /**
-     * Returns the api error description based on the status code.
-     *
-     * @param int $errorCode
-     *
-     * @return string
-     */
-    protected function getApiErrorDocDescriptionByErrorCode($errorCode)
-    {
-        switch ((int)$errorCode) {
-            case 400:
-                return 'Bad request, the request parameters are invalid';
-                
-            case 401:
-                return 'Authorization required for calling this endpoint';
-                
-            case 402:
-                return 'Billing error, payment required';
-                
-            case 403:
-                return 'The authenticated user has no permission for this operation';
-                
-            case 404:
-                return 'Entity not found';
+            case self::SWAGGER_VERSION_V3:
+                $generator     = new OpenApiGenerator();
```

Comment:
I'm sure there will be an open api v4 at some point. Just to save ourselves a headache in the future, could we call this OpenApiV3Generator or something like that?